src/App.tsx

```diff
@@ -1,73 +1,84 @@
+      callback(...args);
+    }, delay);
+  };
+}
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
+export const App: React.FC = () => {
+  const delay = 300;
+  const clearPerson = {
+    name: '',
+    born: 0,
+    died: 0,
+  };
+  const [people] = useState(peopleFromServer);
+  const [searchPeople, setSearchPeople] = useState('');
+  const [apliedPeople, setApliedPeople] = useState('');
+  const [usePeople, setUsePeople] = useState(false);
+  const [activePerson, setActivePerson] = useState(clearPerson);
+  const { name, born, died } = activePerson;
+  const handleUsePeople = () => {
+    setUsePeople(true);
+  };
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
+  const handleClickUser = (person: Person) => {
+    setUsePeople(false);
+    setSearchPeople(person.name);
+    setApliedPeople(person.name);
+    setActivePerson({
+      name: person.name,
+      born: person.born,
+      died: person.died,
+    });
+  };
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyPeople = useCallback(debounse(setApliedPeople, delay), []);
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
+  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchPeople(e.target.value);
+    applyPeople(e.target.value);
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
+    if (e.target.value !== name) {
+      setActivePerson(clearPerson);
+    }
+  };
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
+  const peopleFilter = useMemo(() => {
+    return people.filter(it =>
+      it.name.toLowerCase().includes(apliedPeople.toLowerCase()),
+    );
+  }, [people, apliedPeople]);
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Hercke</p>
-              </div>
-            </div>
-          </div>
-        </div>
+  return (
+    <div className="container">
+      <main className="section is-flex is-flex-direction-column">
+        <h1 className="title" data-cy="title">
+          {name ? `${name} (${born} - ${died})` : 'No selected person'}
```

Comment:
Consider will be better if you create a helper variable for this logic and use it here